src/main/java/duke/handlers/HandlerFactory.java

```diff
@@ -0,0 +1,73 @@
+package duke.handlers;
+
+import duke.enums.Command;
+import duke.exceptions.DukeException;
+import duke.service.Service;
+
+
+public class HandlerFactory {
+    String command;
+    String taskName;
+    String flag;
+    String flagOption;
+
+    private static class UnknownHandler implements IHandler {
+        @Override
+        public void handle(Service s) throws DukeException {
+            throw new DukeException("Unknown command! Please try again.");
+        }
+    }
+
+    public HandlerFactory(String command) {
+        this.command = command;
+    }
+
+    public HandlerFactory taskName(String taskName) {
+        this.taskName = taskName;
+        return this;
+    }
+
+    public HandlerFactory flag(String flag) {
+        this.flag = flag;
+        return this;
+    }
+
+    public HandlerFactory flagOption(String option) {
+        this.flagOption = option;
+        return this;
+    }
+
+    /**
+     * Returns a Hander based on the command input from the user
+     *
+     * @return IHandler
+     */
+    public IHandler build() {
+        try {
+            Command commandEnum = Command.valueOf(this.command.toUpperCase());
+            switch (commandEnum) {

```

Comment:
I like how you abstracted the commands into an enum, improves the readability of your code greatly! 👍